### PR TITLE
feat: support increase l2 tx max cycles

### PR DESCRIPTION
**Pull request by jjyr**
1. add fork config
2. fetch consensus parameters from fork config

---------------

Configuration changes:

1. move `backend_switches` to `fork.backend_forks`
2. move `debug_backend_switches` to `debug_backend_forks`

### CODE REVIEW

File: crates/generator/src/generator.rs [hunk 1]
```diff
@@ -734,6 +742,7 @@ impl Generator {
         let backend = self
             .load_backend(block_info.number().unpack(), state, &script_hash)
```
Comment (blckngm) [inline crates/generator/src/generator.rs:721]: Why not load `max_cycles` here?

Comment (jjyr) [reply crates/generator/src/generator.rs:721]: Debug API support pass `max_cycles` from request. It helps us debug cycles error.

We can remove the argument if we can ensure a polyjuice tx can be 100% verified within the default cycles.

Comment (blckngm) [reply crates/generator/src/generator.rs:721]: How about adding max cycles config to debug backend config?

Comment (blckngm) [reply crates/generator/src/generator.rs:721]: OK. I see that debug max cycles is from RPC params.

So how about taking `override_max_cycles: Option<u64>`, and load from config if it's `None`.

Comment (jjyr) [reply crates/generator/src/generator.rs:721]: Good idea

**Review by keroro520 — DISMISSED** [review]

**Review by Flouse — COMMENTED** [review]
/itest

**Review by Flouse — APPROVED** [review]

**Review by blckngm — APPROVED** [review]

**Pull request merged**